pytest_container/runtime.py

```diff
@@ -280,31 +259,30 @@ def get_image_size(
             else str(image_or_id_or_container)
         )
         return float(
-            check_output(
-                [
-                    self.runner_binary,
-                    "inspect",
-                    "-f",
-                    '"{{ .Size }}"',
-                    id_to_inspect,
-                ]
-            )
-            .decode()
-            .strip()
-            .replace('"', "")
+            self.run_command(
+                "inspect",
+                "-f",
+                '"{{ .Size }}"',
+                id_to_inspect,
+            ).replace('"', "")
         )
 
-    def _get_container_inspect(self, container_id: str) -> Any:
+    def _run_inspect(self, container_id: str) -> Union[Dict[str, Any], None]:
         inspect = json.loads(
-            check_output([self.runner_binary, "inspect", container_id])
+            self.run_command("inspect", container_id, ignore_errors=True)
         )
-        if len(inspect) != 1:
+        if not isinstance(inspect, list):
+            raise RuntimeError(
+                f"Expected a list of results, but got {inspect} instead"
```

Comment:
Did you want to include the type in the error message instead?
```suggestion
                f"Expected a list of results, but got {type(inspect)} instead"
```

Reply:
No, I wanted to result of the command, so it's clear what went wrong.